civo/instances/resource_instance.go

```diff
@@ -468,6 +469,56 @@ func resourceInstanceUpdate(ctx context.Context, d *schema.ResourceData, m inter
 		}
 	}
 
+	// If reserved_ipv4 has changed, update the instance with the new reserved IP
+	if d.HasChange("reserved_ipv4") {
+		oldReservedIP, newReservedIP := d.GetChange("reserved_ipv4")
+		instance, err := apiClient.GetInstance(d.Id())
+		if err != nil {
+			// Check if the instance no longer exists.
+			return diag.Errorf("[ERR] instance %s not found", d.Id())
+		}
+
+		// Unassign the old reserved IP if it exists
+		if oldReservedIP != "" {
+			ip, err := apiClient.FindIP(oldReservedIP.(string))
+			if err != nil {
+				if errors.Is(err, civogo.ZeroMatchesError) {
+					return diag.Errorf("sorry there is no %s IP in your account", oldReservedIP)
+				} else if errors.Is(err, civogo.MultipleMatchesError) {
+					return diag.Errorf("sorry we found more than one IP with that value in your account")
+				} else {
+					return diag.Errorf("error finding IP %s: %s", oldReservedIP, err)
+				}
+			}
+
+			_, err = apiClient.UnassignIP(ip.ID, apiClient.Region)
+			if err != nil {
+				return diag.Errorf("[ERR] an error occurred while unassigning reserved IP %s from instance %s: %s", ip.ID, instance.ID, err)
+			}
+			log.Printf("[INFO] unassigned reserved IP %s from the instance %s", oldReservedIP, d.Id())
```

Comment:
And this one - why "[INFO]" when other messages are not always prefaced with a classification like this?